plugin-server/src/cdp/utils.ts

```diff
@@ -114,14 +134,36 @@ export function convertToHogFunctionFilterGlobal(globals: HogFunctionInvocationG
         }
     }
 
-    return {
+    const elementsChain = globals.event.properties['$elements_chain']
+    const response = {
         event: globals.event.name,
-        elements_chain: globals.event.properties['$elements_chain'],
+        elements_chain: elementsChain,
+        elements_chain_href: '',
+        elements_chain_texts: [],
+        elements_chain_ids: [],
+        elements_chain_elements: [],
         timestamp: globals.event.timestamp,
         properties: globals.event.properties,
         person: globals.person ? { properties: globals.person.properties } : undefined,
         ...groups,
     }
+    if (elementsChain) {
+        Object.defineProperties(response, {
+            elements_chain_href: {
+                get: () => getElementsChainHref(elementsChain),
```

Comment:
Love this solution. I'm wondering though if we should also cache these values? Maybe not worth it for now...

Also a comment here would be helpful for future travellers to understand what this is all for (I barely get it right now)

Reply:
Also its not awesome that there is only a test for one path...

Reply:
I think I found a rather neat way to cache this: https://github.com/PostHog/posthog/pull/24331/commits/e2e1bc73721ad764b9cf9bd63982c1123200886f

More tests coming soon...

Reply:
wooohhhhhh self destructing function :o